client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx

```diff
@@ -157,7 +166,6 @@ const PlansWrapper: React.FC< Props > = ( props ) => {
 					domainName={ getDomainName() }
 					customerType={ customerType }
 					plansWithScroll={ isDesktop }
-					planTypes={ planTypes }
```

Comment:
This is probably the cause for the issue this PR is trying to fix: https://github.com/Automattic/wp-calypso/pull/76657#pullrequestreview-1418592479
The `planTypes` is being used to filter out what plans to show. Unfortunately, this causes issues with the 2023 Pricing grid because this version needs all the plans to compile the available features ([more info](https://github.com/Automattic/wp-calypso/pull/75424)).

This PR fixes the issue by working around the `planTypes` and filtering via individual flags (`hidePersonalPlan`, etc.) but I see a couple of downsides:
* it only works for this instance of `PlansFeaturesMain`
* having a flag for each plan looks hard to maintain
* looks complicated: we have a filtered list of plans (`planTypes`) and construct booleans from each to re-create it into another list (`visiblePlans`)

I propose another approach in this PR: https://github.com/Automattic/wp-calypso/pull/76763